src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+                        yield return analyzer;
+                    }
+
+                    continue;
+                }
+                else if (analyzer.SupportedDiagnostics.Any(y => options.AnalyzerIds.Contains(y.Id)))
+                {
+                    yield return analyzer;
+                }
+            }
+        }
+
+        private static ImmutableArray<Assembly> GetAnalyzerAssemblies(string path)
+        {
+            if (File.Exists(path))
+            {
+                return ImmutableArray.Create(Assembly.LoadFrom(path));
+            }
+            else if (Directory.Exists(path))
+            {
+                var builder = ImmutableArray.CreateBuilder<Assembly>();
+                foreach (var file in Directory.GetFiles(path, "*.dll", SearchOption.AllDirectories))
+                {
+                    builder.Add(Assembly.LoadFrom(file));
+                }
+
+                return builder.ToImmutable();
+            }
+
+            Utilities.WriteLine($"Cannot load assembliy or assemblies from {path}.", ConsoleColor.Red);
+            return ImmutableArray<Assembly>.Empty;
+        }
+
+        private static IEnumerable<Type> GetLoadableTypes(Assembly assembly)
+        {
+            if (assembly == null) throw new ArgumentNullException(nameof(assembly));
+            try
+            {
+                return assembly.GetTypes();
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                Console.WriteLine($"Failed to load all types from {assembly.FullName}. Will proceed with types loaded successfully.");
+                return e.Types.Where(t => t != null);
+            }
+        }
+
+        private static ImmutableArray<DiagnosticAnalyzer> GetAllAnalyzers(IEnumerable<Assembly> assemblies)
+        {
+            var analyzers = ImmutableArray.CreateBuilder<DiagnosticAnalyzer>();
+            foreach (var diagnosticAnalyzer in GetAllSubclasses<DiagnosticAnalyzer>(assemblies))
+            {
+                analyzers.Add(diagnosticAnalyzer);
+            }
+
+            return analyzers.ToImmutable();
+        }
+
+        private static ImmutableDictionary<string, ImmutableList<CodeFixProvider>> GetAllCodeFixers(IEnumerable<Assembly> assemblies)
+        {
+            Dictionary<string, ImmutableList<CodeFixProvider>> providers = new Dictionary<string, ImmutableList<CodeFixProvider>>();
+            foreach (var codeFixProvider in GetAllSubclasses<CodeFixProvider>(assemblies))
+            {
+                foreach (var diagnosticId in codeFixProvider.FixableDiagnosticIds)
+                {
+                    providers.AddToInnerList(diagnosticId, codeFixProvider);
+                }
+            }
+
+            return providers.ToImmutableDictionary();
+        }
+
+        private static IEnumerable<T> GetAllSubclasses<T>(IEnumerable<Assembly> assemblies)
+        {
+            var returnType = typeof(T);
+            foreach (var assembly in assemblies)
```

Comment:
hmmm... it doesn't check attribute to discover. rather it is using type to discover diagnostic analyzer. wonder how this filter through to right language. 

why not just use this (http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalyzerReference.cs,59)
create var analyzerReference = new AnalyzerFileReference(path to analyzer file, AssemblyLoader.Default)

and you can create AssemblyLoader from this
http://source.roslyn.io/#Roslyn.Compilers.UnitTests/AnalyzerFileReferenceTests.cs,19